packages/flame/lib/src/collisions/quadtree/broadphase.dart

```diff
@@ -0,0 +1,161 @@
+import 'dart:collection';
```

Comment:
Since there is another broadphase too I wouldn't name this file `broadphase`, but something more specific.
And maybe we can create a directory to keep the broadphase systems in, and then you can move out the mixins so that they are next to the other mixins in `collisions` (or possibly make a new directory for that too if you think that would be cleaner).

Reply:
I usually try to deattach different implementations of an basic class from each other. But here we have multiple classes and mixins for each realisation. From my point of view it would be better to: 
 - move all Sweep-related classes/mixins to corresponding dir
 - move all Quadtree-related stuff to another dir
 - put "core" collision libraries somwhere else, maybe just keep at root of collision subdir 

I did not moved sweep files in my PR to avoid problems with branch updating from upstream while working on PR. Just dont like merge conflicts 🙃

Reply:
I would keep the actual collision detection systems directly in `collisions/` (including `QuadTreeCollisionDetection`) and then the files relating to the broadphase in `collisions/broadphase` and then the specific implementations in `collisions/broadphase/sweep` and `collisions/broadphase/quadtree`.

> I did not moved sweep files in my PR to avoid problems with branch updating from upstream while working on PR. Just dont like merge conflicts 🙃

We can move those in a separate PR. :)

Reply:
Yes, I'd prefer to do this in last turn, when all code changes will be performed  👍🏻

Reply:
@spydon  I just opened new PR about it: https://github.com/flame-engine/flame/pull/1943#issue-1388132672